modules/color-picker/react/lib/ColorInput.tsx

```diff
@@ -44,26 +49,56 @@ const CustomHexInput = styled(TextInput)<Pick<ColorInputProps, 'disabled' | 'gro
     },
   ({disabled}) => ({
     backgroundColor: disabled ? colors.soap200 : '',
+  }),
+  ({theme}) => ({
+    paddingLeft: theme.direction === ContentDirection.LTR ? '46px' : 'calc(100% - 86px) /* @noflip */' ,
   })
 );
 
-const ColorInputContainer = styled('div')({
-  position: 'relative',
-});
+// Note regarding:
+//   ({theme}) => ({
+//     paddingLeft: theme.direction === ContentDirection.LTR ? '46px' : 'calc(100% - 86px) /* @noflip */' ,
+//   })
+//
+// We're using @noflip because ColorInput should stay LTR, therefore, we need to adjust the padding using ContentDirection, not using rtl-css-js.
+
+const ColorInputContainer = styled('div')<Pick<ColorInputProps, 'grow'>>(
+  {
+    position: 'relative',
+    width: colorInputWidth,
+  },
+  ({grow}) =>
+    grow && {
+      minWidth: '100%',
+      width: '100%',
+    },  
+);
 
 const PoundSignPrefix = styled('span')<Pick<ColorInputProps, 'disabled'>>(
   {
     position: 'absolute',
-    left: 36,
     top: 10,
     ...type.body,
-    ...type.variant.hint,
+    ...type.variant.mono,
+    color: colors.blackPepper300,
   },
   ({disabled}) => ({
     color: disabled ? inputColors.disabled.text : undefined,
+  }),
+  ({theme}) => ({
+    left: theme.direction === ContentDirection.LTR ? '36px' : '88px',
   })
 );
 
+// Note regarding:
+//   left: theme.direction === ContentDirection.LTR ? '36px' : '88px',
+//
+// This returns:
+//    - LTR -> left: 36px;
+//    - RTL -> right: 88px;
+//
+// It's because we're changing the value of this attribute based on ContentDirection but the attribute itself is getting flipped by rtl-css-js.
```

Comment:
Can you just put this inline on line 89/90? Also you're explaining what's going on here, but not _why_ we needed to do it.